types/recordrtc/index.d.ts

```diff
@@ -0,0 +1,181 @@
+// Type definitions for recordrtc 5.6
+// Project: http://RecordRTC.org/
+// Definitions by: Kyle Hensel <https://github.com/k-yle>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type State = 'inactive' | 'recording' | 'stopped' | 'paused' | 'destroyed';
+
+interface Disk {
+    audio?: Blob;
+    video?: Blob;
+    gif?: Blob;
+}
+
+interface Options {
+    type?: 'video' | 'audio' | 'canvas' | 'gif';
+
+    mimeType?:
+        | 'video/webm'
+        | 'audio/webm'
+        | 'video/mp4'
+        | 'video/webm;codecs=vp9'
+        | 'video/webm;codecs=vp8'
+        | 'video/webm;codecs=h264'
+        | 'video/x-matroska;codecs=avc1'
+        | 'video/mpeg'
+        | 'audio/wav'
+        | 'audio/ogg';
+
+    disableLogs?: boolean;
+
+    /** get intervals based blobs value in milliseconds */
+    timeSlice?: number;
+
+    /** requires `timeSlice` to be set */
+    ondataavailable?(cb: (blob: Blob) => void): void;
+
+    /** auto stop recording if camera stops */
+    checkForInactiveTracks?: boolean;
+
+    /** requires timeSlice above */
+    onTimeStamp?(cb: (timestamp: number) => void): void;
+
+    /** both for audio and video tracks */
+    bitsPerSecond?: number;
+
+    /** only for audio track */
+    audioBitsPerSecond?: number;
+
+    /** only for video track */
+    videoBitsPerSecond?: number;
+
+    /** used by CanvasRecorder and WhammyRecorder, it is kind of a "frameRate" */
+    frameInterval?: number;
+
+    /** if you are recording multiple streams into single file, this helps you see what is being recorded */
+    previewStream?(cb: (stream: MediaStream) => void): void;
+
+    /** used by CanvasRecorder and WhammyRecorder */
+    video?: HTMLVideoElement;
+
+    /** used by CanvasRecorder and WhammyRecorder */
+    canvas?: {
+        width: number;
+        height: number;
+    };
+
+    /** used by StereoAudioRecorder, the range is 22050 to 96000 (kHz). */
+    sampleRate?: number;
+
+    /** used by StereoAudioRecorder. the range is 22050 to 96000 (kHz). */
+    desiredSampRate?: number;
+
+    /** used by StereoAudioRecorder */
+    bufferSize?: 256 | 512 | 1024 | 2048 | 4096 | 8192 | 16384;
+
+    /** used by StereoAudioRecorder */
+    numberOfAudioChannels?: 1 | 2;
+
+    /** used by WebAssemblyRecorder */
+    frameRate?: number;
+
+    /** used by WebAssemblyRecorder */
+    bitrate?: number;
+
+    /** used by MultiStreamRecorder - to access HTMLCanvasElement */
+    elementClass?: string;
+}
+
+declare class RecordRTC {
```

Comment:
Moving types except class into namespace RecordRTC will let library users use these types if they need to so thats a better way to organize things.

Reply:
Hi @sheetalkamat, I'm not entirely sure what you mean by that. Are you suggesting something like this?
```ts
namespace RecordRTC {
    type State = 'inactive' | 'recording';

    interface Disk {
        // ...
    }

    // ...
}

declare class RecordRTC {
    constructor();

    // ... 
}

export = RecordRTC;

```